bin/repolinter.js

```diff
@@ -66,42 +65,12 @@ require('yargs')
     },
     async (/** @type {any} */ argv) => {
       let rulesetParsed = null
-      let jsonerror
-      let yamlerror
       // resolve the ruleset if a url is specified
       if (argv.rulesetUrl) {
-        const res = await fetch(argv.rulesetUrl)
-        if (!res.ok) {
-          console.error(
-            `Failed to fetch config from ${argv.rulesetUrl} with status code ${res.status}`
-          )
-          process.exitCode = 1
-          return
-        }
-        const data = await res.text()
-        // attempt to parse as JSON
         try {
-          rulesetParsed = JSON.parse(data)
+          rulesetParsed = await config.loadConfig(argv.rulesetUrl)
```

Comment:
Calling `config.loadConfig` in both the CLI and in `lint` seems a bit much, bit I see why you've done it. It might be cleaner if we decide that the lint API should handle all config resolution, meaning the `rulset` parameter can be a relative path to a file, an absolute path to a file, http/https URL, an object, or `null`: that way we can just pass the `argv.rulesetUrl` directly to the lint API and let `config.loadConfig` handle all the cases (also getting rid of a lot of logic in the `lint` function itself). Does that make sense?

Reply:
removed the `config.loadConfig` call here.  I have not yet merged the `--rulesetFile` and `--rulesetUrl` flags yet, though they are at this point duplicate.